### DISCUSSION TITLE
what are the "worker-xxx.js"?

### DISCUSSION THREAD

**Discussion by kuchaguangjie** [ajaxorg/ace#4855] [Q&A] [answered]
e.g `worker-css.js`

**Comment by nightwing** [comment, marked as answer]
They are linters which are loaded into web worker, and provide annotations on the gutter of the editor

**Reply by kuchaguangjie** [reply]
So, how could I enable that, I've include following scripts, but still not got a hint on symtax issue:
```

  <script src="https://cdnjs.cloudflare.com/ajax/libs/ace/1.8.1/ace.min.js" type="text/javascript" charset="utf-8"></script>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/ace/1.8.1/mode-javascript.min.js" type="text/javascript" charset="utf-8"></script>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/ace/1.8.1/theme-dracula.min.js" type="text/javascript" charset="utf-8"></script>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/ace/1.8.1/ext-error_marker" type="text/javascript" charset="utf-8"></script>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/ace/1.8.1/worker-base.min.js" type="text/javascript" charset="utf-8"></script>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/ace/1.8.1/worker-javascript.min.js" type="text/javascript" charset="utf-8"></script>
```
I've set the mode as javascript.

**Comment by nightwing** [comment]
It appears there is a regression in the minified version, using "https://cdnjs.cloudflare.com/ajax/libs/ace/1.8.1/ace.js" works see 

https://jsbin.com/vosaguzada/edit?html,output
```html
<body>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/ace/1.8.1/ace.js" type="text/javascript" charset="utf-8"></script>
  <script charset="utf-8">
    var editor = ace.edit(null, {
      theme: "ace/theme/monokai",
      mode: "ace/mode/javascript",
    });
    document.body.appendChild(editor.container)
    editor.container.style.height= "100px"
  </script>
</body> 
```

**Reply by kuchaguangjie** [reply]
That works. So, seems there is no need to include other js by hand when use the normal version, they will be auto loaded by `ace.js` as need.
Screenshot in chrome devtools:
  ![ace](https://user-images.githubusercontent.com/4230959/180643609-6b4b91b6-7df7-4ce8-804e-ec2851ecf0a6.jpg)

**Comment by nightwing** [comment]
Ace loads other js when needed both for normal and minified versions.

**Reply by kuchaguangjie** [reply]
It doesn't auto load other js when include `ace.min.js`, as mentioned above.
e.g 
`<script src="https://cdnjs.cloudflare.com/ajax/libs/ace/1.8.1/ace.min.js" type="text/javascript" charset="utf-8"></script>`

Is this going to be fixed?

**Reply by nightwing** [reply]
Yes, it is a regression that will be fixed in 1.8.2

**Reply by kuchaguangjie** [reply]
Seems the issue is it try to load other js relative to current dir, not from the cdn.

**Screenshot:**

   ![ace-load-failed](https://user-images.githubusercontent.com/4230959/180643757-a16b083a-74e9-4f9e-8b5f-aaef51762483.jpg)
